From 25288ce11c6e9a4b777739acae0d5e74f91983be Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Tue, 28 May 2024 12:32:46 +0900
Subject: [PATCH] FIX: operator precedence related to poll
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes the following warnings:

```
connect.c:162:19: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
mailstream_ssl.c:368:32: warning: self-comparison always evaluates to false [-Wtautological-compare]
mailstream_ssl.c:368:32: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
```
---
 src/data-types/connect.c        | 2 +-
 src/data-types/mailstream_ssl.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/data-types/connect.c b/src/data-types/connect.c
index e17a5e03..23a2fd14 100644
--- a/src/data-types/connect.c
+++ b/src/data-types/connect.c
@@ -159,7 +159,7 @@ static int wait_connect(int s, int r, time_t timeout_seconds)
     return -1;
   }
   
-  if (pfd.revents & POLLOUT != POLLOUT) {
+  if ((pfd.revents & POLLOUT) != POLLOUT) {
     return -1;
   }
 #endif
diff --git a/src/data-types/mailstream_ssl.c b/src/data-types/mailstream_ssl.c
index bd680ae5..deb7df7f 100644
--- a/src/data-types/mailstream_ssl.c
+++ b/src/data-types/mailstream_ssl.c
@@ -365,7 +365,7 @@ static int wait_SSL_connect(int s, int want_read, time_t timeout_seconds)
     return -1;
   }
 
-  if (pfd.revents & pfd.events != pfd.events) {
+  if ((pfd.revents & pfd.events) != pfd.events) {
     return -1;
   }
 #endif
